class-instant-articles-signer.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Facebook Instant Articles for WP.
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @package default
+ */
+
+/**
+ * Class responsible for handling the signature of graph rescrape calls
+ *
+ * @since 4.1
+ */
+class Instant_Articles_Signer {
+
+  const PUBLIC_KEY_OPTION = 'instant-articles-rescrape-public-key';
+  const PRIVATE_KEY_OPTION = 'instant-articles-rescrape-private-key';
+  const PUBLIC_KEY_PATH = '.well-known/graph-api/apikey.pub';
+
+  public static function init() {
+    add_action( 'wp', array( 'Instant_Articles_Signer', 'output_public_key' ) );
+  }
+
+  public static function output_public_key() {
+    global $wp;
+    if ( $wp->request == self::PUBLIC_KEY_PATH ) {
+      status_header(200);
+      echo self::get_public_key();
```

Comment:
Shouldn't we add proper content-type?

Reply:
Check if we shouldn't allow more text into this response. So `the_content` should not have anything else but the key.

Reply:
We `die()` aftwards to ensure nothing else is coming in out the response.